_search-plugins/search-pipelines/ml-inference-search-response.md

```diff
@@ -388,4 +388,374 @@ The response confirms that the processor has generated text embeddings in the `p
+
+
+### Example: Rerank Use Case
+
+The following example shows you how to configure an `ml_inference` search response processor with a text similarity model.
+
+
+Step 0: Host a model
+The pre-requisite is a registered genai model in opensearch. For more information about externally hosted models, see [Connecting to externally hosted models]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/index/). Here is a sample predict response using a registered model, which requires a prompt and a context field.
+
+```json
+POST _plugins/_ml/models/tg5p1ZEB4iWlnHsIh2U9/_predict
+{
+    "query_text": "today is sunny",
+    "text_docs": [
+        "how are you",
+        "today is sunny",
+        "today is july fifth",
+        "it is winter"
+    ]
+}
+```
+
+The model returns similarity scores for each input document:
+
+```json
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "similarity",
+          "data_type": "FLOAT32",
+          "shape": [1],
+          "data": [-11.055183]
+        }
+      ]
+    },
+    {
+      "output": [
+        {
+          "name": "similarity",
+          "data_type": "FLOAT32",
+          "shape": [1],
+          "data": [8.969885]
+        }
+      ]
+    },
+    {
+      "output": [
+        {
+          "name": "similarity",
+          "data_type": "FLOAT32",
+          "shape": [1],
+          "data": [-5.736347]
+        }
+      ]
+    },
+    {
+      "output": [
+        {
+          "name": "similarity",
+          "data_type": "FLOAT32",
+          "shape": [1],
+          "data": [-10.0452175]
+        }
+      ]
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+Step 1: Index sample documents
+
+Create an index and add some sample documents:
+
+```json
+PUT /demo-index-0/_doc/1
+{
+  "dairy": "how are you"
+}
+
+PUT /demo-index-0/_doc/2
+{
+  "dairy": "today is sunny"
+}
+
+PUT /demo-index-0/_doc/3
+{
+  "dairy": "today is july fifth"
+}
+
+PUT /demo-index-0/_doc/4
+{
+  "dairy": "it is winter"
+}
+
+```
+{% include copy-curl.html %}
+
+Step 2: Create a search pipeline
+
+Create a search pipeline that uses the text similarity model:
+
+```json
+PUT /_search/pipeline/my_pipeline
+{
+  "response_processors": [
+    {
+      "ml_inference": {
+        "tag": "ml_inference",
+        "description": "This processor runs ml inference during search response",
+        "model_id": "tg5p1ZEB4iWlnHsIh2U9",
+        "model_input": "{ \"text_docs\": ${input_map.text_docs}, \"query_text\": \"${input_map.query_text}\" }",
+        "function_name": "TEXT_SIMILARITY",
+        "input_map": [
+          {
+            "text_docs": "dairy",
+            "query_text": "$.query.term.dairy.value"          
+          }
+        ],
+        "output_map": [
+          {
+            "rank_score": "$.inference_results[*].output[*].data"
+          }
+        ],
+        "full_response_path": false,
```

Comment:
`"$.inference_results[*].output[*].data"` this looks like a fulll response path, should we use `false` here ?

Reply:
I changed the use case to use full_response_path true